contracts/interfaces/modules/metadata/ICoreMetadataModule.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.23;
+
+import { IModule } from "../../../../contracts/interfaces/modules/base/IModule.sol";
+
+/// @title CoreMetadataModule
+/// @notice Manages the core metadata for IP assets within the Story Protocol.
+/// @dev This contract allows setting and updating core metadata attributes for IP assets.
+interface ICoreMetadataModule is IModule {
+    /// @notice Sets the name for an IP asset.
+    /// @dev Can only be called once per IP asset to prevent overwriting.
+    /// @param ipAccount The address of the IP asset.
+    /// @param name The name to set for the IP asset.
+    function setIpName(address ipAccount, string memory name) external;
+
+    /// @notice Sets the description for an IP asset.
+    /// @dev Can only be called once per IP asset to prevent overwriting.
+    /// @param ipAccount The address of the IP asset.
+    /// @param description The description to set for the IP asset.
+    function setIpDescription(address ipAccount, string memory description) external;
+
+    /// @notice Sets the content hash for an IP asset.
+    /// @dev Can only be called once per IP asset to prevent overwriting.
+    /// @param ipAccount The address of the IP asset.
+    /// @param contentHash The content hash to set for the IP asset.
+    function setIpContentHash(address ipAccount, bytes32 contentHash) external;
```

Comment:
`bytes32` might be too limiting for contentHash, since there are hashing schemas that result in payloads > 32 bytes 
We could `keccak256` but we lose info.
My suggestion would be to use either `bytes` or `string` for future proofing

Reply:
Thank you for your feedback on using bytes32 for content hashes. Considering both Web3 (like IPFS and Arweave) and Web2 (Amazon S3 and Google Cloud Storage) storage protocols use 256-bit hashes, our current approach aligns with industry standards. This balance ensures efficiency and compatibility while keeping gas costs low. However, I acknowledge the importance of future-proofing and am open to revisiting this decision as industry standards evolve or our protocol's needs change. Let's continue this discussion to ensure our solution remains both practical and forward-looking.

Reply:
Sounds good!